src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -105,7 +105,7 @@ public TestPlatform()
         loggerManager.Initialize(discoveryCriteria.RunSettings);
 
         var testHostManager = _testHostProviderManager.GetTestHostManagerByRunConfiguration(discoveryCriteria.RunSettings);
-        ThrowExceptionIfTestHostManagerIsNull(testHostManager, discoveryCriteria.RunSettings);
+        TestPlatform.ThrowExceptionIfTestHostManagerIsNull(testHostManager, discoveryCriteria.RunSettings);
```

Comment:
Which rule forces this, and what advantage is it bringing us? Are static methods significantly faster? Will we be forced to mark every method that does not touch class fields as static? Is not touching class fields considered code smell and this rule just makes the smell easier to see?

Reply:
> Which rule forces this, and what advantage is it bringing us?
Are you referring to `static` or the prefix? If that's about the prefix, the code-fix for make member `static` seems to be adding the class before but that's not required for C# so I could simplify all names if we prefer.

> Are static methods significantly faster? 
They are faster but perf is always tricky, it depends on so many things. I just don't see any reason not to benefit from the gain (even if proved small in some context). But I would not worry about breaking public API (I suppressed the warnings for them) only for such small change.

> Will we be forced to mark every method that does not touch class fields as static?
Yes, except if we suppress it. I could keep the rule as suggestion but then it's unlikely to be followed.

> Is not touching class fields considered code smell and this rule just makes the smell easier to see?
I am not sure to understand the question.

Reply:
I was referring to the static, but I did not realize that you are adding the name of the class, because the rule does that. I thought you don't need that, but I was not sure, and did not bother to try it out. I would prefer it that way, so we only call the method by name, not caring if it is static or not. That way making a method static does not result in many changes, and would be much easier to follow.